src/tr1/game/ui/elements/bar.c

```diff
@@ -0,0 +1,135 @@
+#include "game/ui/elements/bar.h"
+
+#include "game/output.h"
+
+#include <libtrx/config.h>
+#include <libtrx/game/scaler.h>
+#include <libtrx/game/ui/helpers.h>
+#include <libtrx/utils.h>
+
+#include <math.h>
+#include <string.h>
+
+#define M_BORDER 2
+#define M_PADDING 2
+#define M_COLOR_STEPS 5
+
+typedef struct {
+    UI_BAR_SETTINGS settings;
+} M_DATA;
+
+#define C(c)                                                                   \
+    { .b = (c) & 0xFF, .g = (c >> 8) & 0xFF, .r = (c >> 16) & 0xFF, .a = 255 }
+static RGBA_8888 m_ColorBarMap[][M_COLOR_STEPS] = {
+    // clang-format off
+    [BC_GOLD]   = { C(0x3D717B), C(0x65929A), C(0x3D717B), C(0x1F5D6B), C(0x004A5B) },
+    [BC_BLUE]   = { C(0x7C5E25), C(0xA1833C), C(0x7C5E25), C(0x644613), C(0x4C2E02) },
```

Comment:
Gold and blue values are swapped here. All of the other colours are fine.

Reply:
Good catch, thanks.